build_runner/test/server/serve_handler_test.dart

```diff
@@ -162,6 +164,97 @@ void main() {
       expect(await response.readAsString(), contains('--track-performance'));
     });
   });
+
+  group('build updates', () {
+    final _entrypointExtensionMarker = '/* ENTRYPOINT_EXTENTION_MARKER */';
+
+    test('injects client code if enabled', () async {
+      _addSource('a|web/some.js', _entrypointExtensionMarker);
+      var response = await serveHandler.handlerFor('web', liveReload: true)(
+          new Request('GET', Uri.parse('http://server.com/some.js')));
+      expect(await response.readAsString(), contains('\$livereload'));
+    });
+
+    test('doesn\'t inject client code if disabled', () async {
+      _addSource('a|web/some.js', _entrypointExtensionMarker);
+      var response = await serveHandler.handlerFor('web', liveReload: false)(
+          new Request('GET', Uri.parse('http://server.com/some.js')));
+      expect(await response.readAsString(), isNot(contains('\$livereload')));
+    });
+
+    test('doesn\'t inject client code in non-js files', () async {
+      _addSource('a|web/some.html', _entrypointExtensionMarker);
+      var response = await serveHandler.handlerFor('web', liveReload: true)(
+          new Request('GET', Uri.parse('http://server.com/some.html')));
+      expect(await response.readAsString(), isNot(contains('\$livereload')));
+    });
+
+    test('doesn\'t inject client code in non-marked files', () async {
+      _addSource('a|web/some.js', 'content');
+      var response = await serveHandler.handlerFor('web', liveReload: true)(
+          new Request('GET', Uri.parse('http://server.com/some.js')));
+      expect(await response.readAsString(), isNot(contains('\$livereload')));
+    });
+
+    test('emmits a message to all listners', () async {
+      Function exposedOnConnect;
+      var mockHandlerFactory = (Function onConnect, {protocols}) {
+        exposedOnConnect = onConnect;
+      };
+      var buildUpdatesWebSocketHandler = BuildUpdatesWebSocketHandler(
+          mockHandlerFactory);
+
+      var streamControllerFirst1 = StreamController<List<int>>();
```

Comment:
nit: can we name these a bit more descriptively?

It is also important to make sure these get closed (in a `tearDown` method ideally).